src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonParameterInfo.cs

```diff
@@ -18,29 +19,109 @@ internal JsonParameterInfo(JsonParameterInfoValues parameterInfoValues, JsonProp
 
+        /// <summary>
+        /// Gets the declaring type of the constructor.
+        /// </summary>
+        public Type DeclaringType => MatchingProperty.DeclaringType;
+
+        /// <summary>
+        /// Gets the zero-based position of the parameter in the formal parameter list.
+        /// </summary>
         public int Position { get; }
+
+        /// <summary>
+        /// Gets the type of this parameter.
+        /// </summary>
+        public Type ParameterType => MatchingProperty.PropertyType;
+
+        /// <summary>
+        /// Gets the name of the parameter.
+        /// </summary>
         public string Name { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the parameter has a default value.
+        /// </summary>
         public bool HasDefaultValue { get; }
 
-        // The default value of the parameter. This is `DefaultValue` of the `ParameterInfo`, if specified, or the `default` for the `ParameterType`.
-        public object? DefaultValue { get; private protected init; }
-        public JsonPropertyInfo MatchingProperty { get; }
-        public bool IsNullable { get; internal set; }
+        /// <summary>
+        /// Gets a value indicating the default value if the parameter has a default value.
+        /// </summary>
+        public object? DefaultValue { get; }
 
-        public Type DeclaringType => MatchingProperty.DeclaringType;
-        public Type ParameterType => MatchingProperty.PropertyType;
-        public JsonConverter EffectiveConverter => MatchingProperty.EffectiveConverter;
-        public bool IgnoreNullTokensOnRead => MatchingProperty.IgnoreNullTokensOnRead;
-        public JsonSerializerOptions Options => MatchingProperty.Options;
+        /// <summary>
+        /// The default value to be passed to the constructor argument array, replacing null with default(TParameter).
+        /// </summary>
+        internal object? EffectiveDefaultValue { get; private protected init; }
+
+        /// <summary>
+        /// Gets a value indicating whether the constructor parameter is annotated as nullable.
+        /// </summary>
+        /// <remarks>
+        /// Contracts originating from <see cref="DefaultJsonTypeInfoResolver"/> or <see cref="JsonSerializerContext"/>,
+        /// derive the value of this parameter from nullable reference type annotations, including annotations
+        /// from attributes such as <see cref="AllowNullAttribute"/> or <see cref="DisallowNullAttribute"/>.
+        ///
+        /// This property has no effect on deserialization unless the <see cref="JsonSerializerOptions.RespectNullableAnnotations"/>
+        /// property has been enabled, in which case the serializer will reject any <see langword="null"/> deserialization results.
+        ///
+        /// This setting is in sync with the associated <see cref="JsonPropertyInfo.IsSetNullable"/> property.
+        /// </remarks>
+        public bool IsNullable => MatchingProperty.IsSetNullable;
```

Comment:
Get and set can have different nullability. Does that matter?

Reply:
It shouldn't, get nullability governs serialization and set nullability governs deserialization. Setters and corresponding constructor parameters can have different nullability, the value that does get picked depends on whether a constructor is being used or not. Otherwise though the two values are always kept in sync.